.github/workflows/delete_test_reports.yml

```diff
@@ -0,0 +1,51 @@
+name: Delete playwright test reports
+on: delete
+
+jobs:
+    delete_reports:
+        if: github.event.ref_type == 'branch'
+        name: Delete Playwright Test Reports
+        runs-on: ubuntu-latest
+        env:
+            # Folder on gh-pages branch that contains all test reports for another branch
+            BRANCH_REPORTS_DIR: reports/${{ github.event.ref }}
+        steps:
+            - uses: actions/checkout@main
+              with:
+                ref: gh-pages
+                token: ${{ secrets.GITHUB_TOKEN }}
+            - name: Set Git User
+              run: |
+                git config --global user.email "github-action@example.com"
+                git config --global user.name "GitHub Action"
+            - name: Check for workflow reports
+              id: has-reports
+              run: |
+                if [ -z "$(ls -A $BRANCH_REPORTS_DIR)" ]; then
+                    echo "BRANCH_REPORTS_EXIST=false" >> $GITHUB_OUTPUT
+                else
+                    echo "BRANCH_REPORTS_EXIST=true" >> $GITHUB_OUTPUT
+                fi
+            - name: Delete reports from repo for branch
+              if: ${{ steps.has-reports.outputs.BRANCH_REPORTS_EXIST == 'true' }}
+              timeout-minutes: 3
+              run: |
+                cd $BRANCH_REPORTS_DIR/..
+        
+                rm -rf ${{ github.event.ref }}
+                git add .
+                git commit -m "workflow: remove all reports for branch ${{ github.event.ref }}"
+        
+                while true; do
```

Comment:
Why the while loop?

Reply:
This is to cater for the scenario during a rebase, another pr action job pushed to the gh-pages branch. This means the push on the current job run will fail. If that's the case and the exit code is non-zero, then we will rebase again and push. Does that make sense or do you see some flaws in the logic?

Reply:
Makes sense, maybe add the comment above the loop?

Reply:
I don't think you can comment in a run command, maybe I just echo an explanation?

Reply:
It should work if you start the line with a `#`, as that is a comment in a shell as well.